src/types.ts

```diff
@@ -1558,8 +1558,12 @@ export class ZodObject<
     const { status, ctx } = this._processInputParams(input);
 
     const { shape, keys: shapeKeys } = this._getCached();
-    const dataKeys = util.objectKeys(ctx.data);
-    const extraKeys = dataKeys.filter((k) => !shapeKeys.includes(k));
+    const extraKeys: string[] = [];
+    for (const key in ctx.data) {
+      if (!shapeKeys.includes(key)) {
```

Comment:
Would it make a difference to use a `Set` for the shape keys? Should be faster than `Array.prototype.includes`

Reply:
I tried and that and it was hard to produce any consistent improvement. I suspect that it's similar perf for smallish objects but faster for objects with lots of keys? I'll try adding a very wide object to the realworld bench to see.

Reply:
Yeah, I suspect homgeneous array include calls are pretty optimized, but Set certainly makes sense semantically. Not a blocker here!

Reply:
I’ve found this benchmark for this exact comparison. Maybe it helps with a future PR for this hotspot: <https://jsbench.me/irkhdxnoqa/2>